lighthouse-core/config/config-plugin.js

```diff
@@ -141,6 +146,49 @@ class ConfigPlugin {
     };
   }
 
+
+  /**
+   * Extract and validate groups JSON added by the plugin.
+   * @param {unknown} groupsJson
+   * @param {string} pluginName
+   * @return {Record<string, LH.Config.GroupJson>|undefined}
+   */
+  static _parseGroups(groupsJson, pluginName) {
+    if (groupsJson === undefined) {
+      return undefined;
+    }
+
+    if (!isObjectOfUnknownProperties(groupsJson)) {
+      throw new Error(`${pluginName} groups json is not valid.`);
+    }
+
+    const groups = Object.entries(groupsJson);
+    if (!groups.length) {
+      throw new Error(`${pluginName} has an empty groups object.`);
```

Comment:
do we care if it's empty? Seems ok to support it for e.g. putting in a placeholder `{}` while creating the plugin

Reply:
Agreed, removed.